packages/core/src/sync-store/postgres/encoding.ts

```diff
@@ -134,6 +134,7 @@ type LogsTable = {
   topic3: Hex | null;
 
   chainId: number;
+  checkpoint?: string;
```

Comment:
Why optional?

Reply:
the checkpoint column is nullable b/c we're overloading this table by also using it to track the factory logs which don't have checkpoints yet. the idea is to move those to a separate table (or possibly something else) and then we can make checkpoint non-null